src/sdk/refundPegout.ts

```diff
@@ -1,12 +1,25 @@
 import { assertTruthy, throwErrorIfFailedTx, validateRequiredFields } from '@rsksmart/bridges-core-sdk'
 import { type PegoutQuote, pegoutQuoteDetailRequiredFields, pegoutQuoteRequiredFields } from '../api'
-import { type LiquidityBridgeContract } from '../blockchain/lbc'
+import { type FlyoverSDKContext } from '../utils/interfaces'
+import { isPegoutRefundable } from './isPegoutRefundable'
+import { FlyoverError } from '../client/httpClient'
 
-export async function refundPegout (quote: PegoutQuote, lbc: LiquidityBridgeContract): Promise<string> {
+export async function refundPegout (quote: PegoutQuote, context: FlyoverSDKContext): Promise<string> {
   validateRequiredFields(quote, ...pegoutQuoteRequiredFields)
   validateRequiredFields(quote.quote, ...pegoutQuoteDetailRequiredFields)
 
-  const result = await lbc.refundPegout(quote)
+  const isRefundable = await isPegoutRefundable(context, quote)
+  if (!isRefundable.isRefundable) {
+    throw new FlyoverError({
+      timestamp: Date.now(),
+      recoverable: false,
+      message: 'Quote is not refundable',
+      details: isRefundable.error
+    })
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const result = await context.lbc!.refundPegout(quote)
```

Comment:
Risk if there is no contract better to show the explicit error

Reply:
Changed.